shared/reducers/tracker.js

```diff
@@ -1,842 +1,399 @@
+  action: TrackerGen.Actions
+): Types.State {
   switch (action.type) {
-    case Constants.showNonUser:
-      if (action.error) {
-        return state
-      }
-
+    case TrackerGen.resetStore:
       return {
         ...state,
-        closed: false,
-        hidden: false,
-        inviteLink: action.payload.throttled ? null : action.payload.inviteLink,
-        isPrivate: action.payload.isPrivate,
-        name: action.payload.assertion,
-        reason: `You opened ${action.payload.folderName}`,
-        serviceName: action.payload.socialAssertion.service,
+        trackers: {},
+        nonUserTrackers: {},
       }
-    case Constants.onClose:
+    case TrackerGen.setStartTimer:
       return {
         ...state,
-        closed: true,
-        hidden: true,
+        timerActive: state.timerActive + 1,
       }
-    default:
-      return state
-  }
-}
-
-function dedupeProofs(proofs: Array<Constants.Proof>): Array<Constants.Proof> {
-  return uniqBy(proofs, 'id')
-}
-
-function updateUserState(
-  username: string,
-  state: Constants.TrackerState,
-  action: Action
-): Constants.TrackerState {
-  switch (action.type) {
-    case Constants.identifyStarted:
-      return {...state, error: null}
-    case Constants.updateReason:
-      // In case the reason is null, let's use our existing reason
+    case TrackerGen.stopTimer:
       return {
         ...state,
-        reason: (action.payload && action.payload.reason) || state.reason,
+        timerActive: state.timerActive - 1,
       }
-    case Constants.updateTrackToken:
+    case TrackerGen.cacheIdentify: {
+      const {goodTill, uid} = action.payload
       return {
         ...state,
-        trackToken: action.payload && action.payload.trackToken,
+        cachedIdentifies: {
+          ...state.cachedIdentifies,
+          [uid]: goodTill,
+        },
       }
-    case Constants.onClose:
+    }
+    case TrackerGen.pendingIdentify: {
+      const {username, pending} = action.payload
       return {
         ...state,
-        closed: true,
-        hidden: false,
-        lastAction: null,
-        needTrackTokenDismiss: !state.trackToken, // did we have a track token at this time?
-        shouldFollow: false, // don't follow if they close x out the window
+        pendingIdentifies: {
+          ...state.pendingIdentifies,
+          [username]: pending ? true : undefined,
+        },
       }
-    case Constants.setNeedTrackTokenDismiss:
+    }
+    case TrackerGen.setRegisterIdentifyUi: {
+      const {started} = action.payload
       return {
         ...state,
-        needTrackTokenDismiss: action.payload.needTrackTokenDismiss,
+        serverStarted: started,
       }
-    case Constants.onWaiting:
-      return {
-        ...state,
-        waiting: action.payload.waiting,
+    }
+
+    case TrackerGen.showNonUser: {
+      const {nonUser} = action.payload
+      return updateNonUserState(state, action.payload.username, state => {
```

Comment:
a bit confusing to have `state` mean two different things so close to each other, but changing either will cause more changes. I guess this'll be better once these are all separate closures